brick_breaker/step_08/lib/src/components/brick.dart

```diff
@@ -24,13 +24,11 @@ class Brick extends RectangleComponent
   void onCollisionStart(
       Set<Vector2> intersectionPoints, PositionComponent other) {
     super.onCollisionStart(intersectionPoints, other);
-    if (other is Ball) {
-      removeFromParent();
+    removeFromParent();
 
-      if (game.world.children.whereType<Brick>().length == 1) {
-        game.world.removeAll(game.world.children.whereType<Ball>());
-        game.world.removeAll(game.world.children.whereType<Bat>());
-      }
+    if (game.world.children.whereType<Brick>().length == 1) {
```

Comment:
```suggestion
    if (game.world.children.query<Brick>().length == 1) {
```
`query` is a cheaper operation on the `ComponentSet` since it is cached